### PR TITLE
Feature/access menu keybinds

### PR DESCRIPTION
**Pull request by BrailleBennett**
[//]: # (This changelog will be part of the release notes)
[//]: # (Please keep anything that isn't part of the changelog above this line and delete unused headings)
[//]: # (Changelog entries should be formatted as unordered lists to ensure consistency when they are collated)

# Changelog

## New Features
- Each function in the access menu can now be bound as a key on your keyboard (all unbound by default aside from narrate target)
## Feature Updates
- The narration menu function key key is now deprecated, and has been replaced by the narrate target key as each thing can be bound independently
- The narrator menu was renamed to the access menu to avoid confusion with the existing narrator feature in Microsoft products
## Bug Fixes
- You can now hold down keys without them speaking their output over and over

### CODE REVIEW

**Comment by BrailleBennett** [comment]
https://github.com/khanshoaib3/minecraft-access-i18n/pull/41 must be merged prior to this PR.

**Comment by BrailleBennett** [comment]
https://github.com/khanshoaib3/minecraft-access-i18n/pull/43 must be merged prior to this PR.

File: docs/keybindings.md
Comment (boholder) [inline docs/keybindings.md:1]: Please reformat the tables

File: docs/config.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ or click `Reset Config` button in the `Config Menu`.
 7. [Health n Hunger](#health-n-hunger)
 8. [Player Warnings](#player-warnings)
 9. [Fall Detector](#fall-detector)
-10. [Narrator Menu](#narrator-menu)
+10. [Narrator Menu](#access-menu)
```
Comment (boholder) [inline docs/config.md]: `Narrator Menu` should be renamed to `Access Menu`, so does the corresponding title, or the `#access-menu` link won't work

File: docs/features.md [hunk 2]
```diff
@@ -174,10 +174,10 @@ It will play a sound effect for every location that meets the set threshold dist
 
 See also: [Configuration](/docs/config.md#fall-detector)
 
-## Narrator Menu
+## Access Menu
 
-We also call it the `F4 Menu` because it requires pressing the `F4` key to open.
-This menu integrates a number of helper functions, and you can execute them by opening the menu or by using the hotkey directly.
+This menu can be accessed by pressing the F4 key by default, and integrates a number of helper functions which can be executed from wintin the menu, or by using the individual hotkey directly.
```
Comment (boholder) [inline docs/features.md]: Typo: `wintin`, `asside`

File: docs/changelog.md
Comment (boholder) [inline docs/changelog.md:1]: Changelog file shouldn't be changed, BTW the links are broken, I don't know when.

File: common/src/main/java/com/github/khanshoaib3/minecraft_access/features/access_menu/accessMenuGUI.java
Comment (boholder) [inline common/src/main/java/com/github/khanshoaib3/minecraft_access/features/access_menu/accessMenuGUI.java:1]: First letter of class name should be uppercase

File: common/src/main/java/com/github/khanshoaib3/minecraft_access/features/access_menu/AccessMenu.java [hunk 3]
```diff
@@ -112,24 +121,43 @@ public void update() {
...
             boolean isF3KeyNotPressed = !KeyUtils.isF3Pressed();
 
-            if (menuKey.isPressing() && hotKey.canBeTriggered()) {
-                // Executes immediately, but will not execute twice until release and press the key again
-                hasFunctionSwitchedBeforeF4Released = true;
-                switchHotKeyFunction();
-            } else if (hotKey.canBeTriggered()) {
-                executeHotKeyFunction();
-            } else if (menuKey.canOpenMenu() && isF3KeyNotPressed && !hasFunctionSwitchedBeforeF4Released) {
+if (!menuKey.isPressing() && menuKey.canOpenMenu() && isF3KeyNotPressed) {
```
Comment (boholder) [inline common/src/main/java/com/github/khanshoaib3/minecraft_access/features/access_menu/AccessMenu.java]: format this line

File: common/src/main/java/com/github/khanshoaib3/minecraft_access/features/access_menu/AccessMenu.java [hunk 4]
```diff
@@ -112,24 +121,43 @@ public void update() {
...
-                // To make the narrator menu open AFTER release the F4 key
-                openNarratorMenu();
+                // To make the access menu open AFTER release the F4 key
+                openAccessMenu();
             }
 
-            if (menuKey.isReleased()) {
-                hasFunctionSwitchedBeforeF4Released = false;
-            }
+            if (KeyBindingsHandler.getInstance().narrateTarget.isPressed() && narrateTargetKey.canBeTriggered()) getBlockAndFluidTargetInformation();
```
Comment (boholder) [inline common/src/main/java/com/github/khanshoaib3/minecraft_access/features/access_menu/AccessMenu.java]: `KeyBindingsHandler.getInstance()...isPressed()` can be omitted, `canBeTriggered()` method does same job

File: common/src/main/java/com/github/khanshoaib3/minecraft_access/features/access_menu/accessMenuGUI.java
Comment (boholder) [inline common/src/main/java/com/github/khanshoaib3/minecraft_access/features/access_menu/accessMenuGUI.java:1]: Help me replace the head of `init` method with `super.init()`

**Review by boholder — CHANGES_REQUESTED** [review]

**Review by boholder — APPROVED** [review]

**Comment by khanshoaib3** [comment]
\[Fast Forward CI\] feature/access-menu-keybinds merged into 1.21...\nNo changelogs found... Pushed the changes to origin.

**Pull request merged**